### DISCUSSION TITLE
Stage manager and external display iPadOS 17.4

### DISCUSSION THREAD

**Discussion by rothgar** [blinksh/blink#1996] [Support] [answered]
### Checklist

- [x] Using latest version on the [App Store](http://itunes.apple.com/app/id1156707581)
- [x] Read the [docs](http://blink.sh/docs/)
- [x] Searched for [existing GitHub issues](https://github.com/blinksh/blink/issues)

### Configuration
I have stage manager turned off and split view turned on
![Screenshot 2024-03-15 at 15 08 40](https://github.com/blinksh/blink/assets/371796/23960899-58d8-4abe-b8c0-db483a90d74d)

### Describe the bug
When plugging in an external display (via USB-C to HDMI) the external display shows a new terminal window and I cannot switch to it (command-o or command-shift-o). If I toggle stage manager on it will allow me to switch to the external window.

I tried switching my blink settings to none and stage and both did the same thing. Other docs and issues mentioned a "mirror" setting but I don't have that in my version (17.2.1.866)

![Screenshot 2024-03-15 at 15 12 37](https://github.com/blinksh/blink/assets/371796/4911308d-552f-4ac5-8328-d3aa7ed6333d)

To get this to work I had turn on stage manager and that allowed me to focus/switch to the external display.

My ultimate goal is to use the external display in mirror mode (for public speaking) but the only way to make that work was to go into settings and turn off all multitasking. This forced blink to go into mirror mode on the external display. I'm not sure why mirroring doesn't work with stage manager turned off and Blink settings set to none.

I think this is a bug and not a misconfiguration but please let me know if there's something else I've missed.

**Comment by carloscabanero** [comment, marked as answer]
Hi! Thanks a lot for reaching out. I know this is a little confusing, but there are a lot of modes, not even properly explained in the iOS APIs, and surprisingly Blink is one of the few apps to still make use of all of them.

First, for what you are trying to do, you need to set the mode to "Stage". "Stage" is really called by iOS APIs "mirror" mode. But not because it is the "mirror" mode that you are trying to do here, but because it mirrors the configuration of the external display. So if you set "Stage" and "Stage Manager" is on, then you will see it as Stage Manager. But if your mode is Mirror mode, then it will follow that mode too.

You will think the proper thing to do is to rename that mode to "Mirror" mode then. But most of the users do Stage Manager and calling that Mirror was confusing, so we renamed that on devices supporting "Stage Manager" to "Stage Mode" and made it the new default.

Now, the other modes are "Projection modes". When selected, those will control how the iPad's "projection" (not the external display) should work. So you can have Stage Manager, but Blink still working on "Projection Mode". And the different options (Scale, Inset, None), control how the edges of the "projection" are rendered.

Thinking this again, probably the right interface is to do a "Projection Mode" on/off switch, instead of External display. That will guarantee that if projection mode is off, we will follow what the device is supposed to do, which would have fixed Stage Manager and your Split View scenario. Then with Projection Mode on, we can show the different insets for the edges. And only enable "Projection Mode" by default in devices that do not have Stage Manager.

**Reply by rothgar** [reply]
Thanks for all the details. Let me try it again with the settings you suggested and see how it works.

**Reply by rothgar** [reply]
Just to follow up. I confirmed everything is working as expected with split view enabled in ipadOS and stage enabled in blink.

**Reply by carloscabanero** [reply]
Glad that worked! I will add a note so when we separate Appearance, we create this as "Projection Mode" instead, as that seems to be more clear.
